File: polocloud-node/src/main/java/dev/httpmarco/polocloud/node/terminal/commands/CommandContext.java
package dev.httpmarco.polocloud.node.terminal.commands;

import lombok.extern.log4j.Log4j2;
import org.jetbrains.annotations.Contract;
import org.jetbrains.annotations.NotNull;

import java.util.HashMap;
import java.util.Map;

@Log4j2
public final class CommandContext {

    private final Map<String, Object> contexts = new HashMap<>();

    @Contract(pure = true)
    @SuppressWarnings("unchecked")
    public <T> T arg(@NotNull CommandArgument<T> argument) {
        return (T) contexts.get(argument.key());
    }

    public void append(@NotNull CommandArgument<?> argument, Object value) {
        this.contexts.put(argument.key(), value);
    }

    @SuppressWarnings("unchecked")
    public <T> T arg(String id) {
        return (T) contexts.get(id);
    }
}

File: agent/src/main/java/dev/httpmarco/polocloud/agent/runtime/local/terminal/commands/CommandContext.kt
package dev.httpmarco.polocloud.agent.runtime.local.terminal.commands

class CommandContext {
    private val contexts = HashMap<String, Any?>()

    @Suppress("UNCHECKED_CAST")
    fun <T> arg(argument: CommandArgument<T>): T {
        return contexts[argument.key] as T
    }

    fun append(argument: CommandArgument<*>, value: Any?) {
        this.contexts.put(argument.key, value)
    }
}